### PR TITLE
do not show coinbase for unsupported countries

### LINKED ISSUE #4292: UX: Strange label “Need to buy Bitcoin outside of the USA?”
“Need to buy Bitcoin outside the USA?” — Well, I don’t know. You tell me!

See the attached screenshot. The label on the first option makes sense to me because I know that it shows something else in the USA. To anybody else, however, the wording is very strange. It also doesn’t really explain what is going on in this dialog nor guide the user towards completing the required steps.

I suggest it be changed to something like “Start by purchasing some BitCoins from an exchange, and then transfer your BitCoins to the wallet address shown below. We recommend so-and-so.  Visit so-and-so.”

It should also mention an amount of BitCoins to purchase and how much that would cost in a local currency (or EUR?). These numbers are important to make the user feel more familiar with BitCoins and their worth in a currency they can relate to. I suggest recommending that the user start with 3 times the minimal monthly contribution so the user will be set up for the first three months. This helps reduce the transaction costs involved with currency exchanges and commits the user to three months. “We recommend you purchase 0,0009 BitCoins (15 EUR) at so-and-so, but you can get started with Brave Payments for as little as  0,0003 BitCoins (5 EUR) per month.” 

The headline is also off, but I’m not sure what can be done to clarify it. (The first option doesn’t transfer funds to Brave, and the second and third option is really the same option.) Maybe “Add BitCoins to your Brave Wallet to get started”

screenshot

### PR DESCRIPTION
**Pull request by jkup**
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

Fix #4292 

Test Plan:

Open Brave from a country where coinbase is not supported. I set my VPN to Israel. Then make sure the following changes occur in the add funds dialog.

+ Title goes from "Three ways to add.." to just "Add Bitcoin"
+ Coinbase logo disappears from the bottom
+ Click QR code and make sure coinbase logo is also gone from that footer.

### CODE REVIEW

File: js/about/preferences.js [hunk 1]
```diff
@@ -300,9 +301,13 @@ class BitcoinDashboard extends ImmutableComponent {
...
-      <div id='coinbaseLogo' />
-      <div id='appstoreLogo' />
-      <div id='playstoreLogo' />
+      {
+        coinbaseCountries.indexOf(this.props.ledgerData.get('countryCode')) > -1
+        ? <div id='coinbaseLogo' />
+        : null
+      }
+      <a href='https://itunes.apple.com/us/app/coinbase-bitcoin-wallet/id886427730?mt=8' target='_blank' id='appstoreLogo' />
+      <a href='https://play.google.com/store/apps/details?id=com.coinbase.android' target='_blank' id='playstoreLogo' />
```
Comment (diracdeltas) [inline js/about/preferences.js]: why are we linking to the coinbase app for countries where coinbase is not supported?

File: js/about/preferences.js [hunk 2]
```diff
@@ -376,11 +381,13 @@ class BitcoinDashboard extends ImmutableComponent {
...
-      <Button l10nId='done' className='pull-right whiteButton' onClick={this.props.hideParentOverlay} />
-    </div>
+    if (coinbaseCountries.indexOf(this.props.ledgerData.get('countryCode')) > -1) {
+      return <div className='panelFooter'>
+        <div id='coinbaseLogo' />
+        <span className='coinbaseMessage' data-l10n-id='coinbaseMessage' />
+        <Button l10nId='done' className='pull-right whiteButton' onClick={this.props.hideParentOverlay} />
+      </div>
+    }
   }
```
Comment (diracdeltas) [inline js/about/preferences.js]: should return null otherwise

File: js/about/preferences.js (hunk 1 above)
Comment (jkup) [inline js/about/preferences.js]: sorry! fixed.

**Comment by diracdeltas** [comment]
when my country code is FR, it tells me to buy btc through cubits but still shows the coinbase logo and footer.
screen shot 2016-10-10 at 1 55 23 pm

**Comment by diracdeltas** [comment]
when my country code is `MD`, the coinbase buy button is there, but clicking it returns a 403 from buy.coinbase.com and does nothing. the same thing happens on master FWIW. i think the list of country codes in this PR and the list of supported countries returned by ledger is not the same.

@mrose17

**Comment by jkup** [comment]
@diracdeltas hmm that's a good catch. Perhaps there is something wrong with our approach. @mrose17 found a list of "supported countries" here https://www.coinbase.com/global but they don't all seem to be supported.

**Comment by diracdeltas** [comment]
2 options:
1. assume js/constants/coinbaseCountries is authoritative and change `BitcoinDashboard.exchangePanel` accordingly
2. assume ledgerInfo.exchangeInfo is authoritative and change the footer logo/links accordingly

**Comment by mrose17** [comment]
@jkup - i agree with @diracdeltas

**Comment by jkup** [comment]
@mrose17 @diracdeltas just to make sure I'm clear. We should do away with all of the checks for the user being in the US and only have whether or not they are in constants/coinbaseCountries?

**Comment by diracdeltas** [comment]
did i understand correctly today that the widget only works in the US but the coinbase apps work elsewhere? if so, this at least should change: https://github.com/brave/browser-laptop/blob/f4418d64977dedb564c4be755e4de30f921ab71b/js/about/preferences.js#L353.

**Comment by jkup** [comment]
@diracdeltas I had a misunderstanding with the logic, should all be working now!

**Comment by diracdeltas** [comment]
lgtm

**Pull request merged**